### PR TITLE
Fix #1777: better error messages on post upload failure

### LINKED ISSUE #1777: Improve error messages on failed post upload
When a post failed to upload, we're displaying the error coming from the server, which is very cryptic for the user and doesn't help him to fix the problem.

### PR DESCRIPTION
**Pull request by maxme**
Fix #1777: better error messages on post upload failure

cc @tonyr59h 

Note: we can't do much better on self hosted sites (we can show the error provided by the API).

### CODE REVIEW

File: WordPress/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1062,15 +1062,16 @@
     <string name="error_edit_comment">An error occurred while editing the comment</string>
     <string name="error_publish_empty_post">Can\'t publish an empty post</string>
     <string name="error_publish_no_network">Can\'t publish while there is no connection. Saved as draft.</string>
-    <string name="error_upload">An error occurred while uploading the %s</string>
+    <string name="error_upload">Error while uploading the %1$s: %2$s</string>
     <string name="error_media_insufficient_fs_permissions">Read permission required on media file</string>
     <string name="error_media_not_found">Media could not be found</string>
     <string name="error_media_unauthorized">You don\'t have permission to view or edit media</string>
-    <string name="error_media_parse_format">Could not parse response to %s media</string>
+    <string name="error_media_parse_format">Could not parse response after media upload</string>
```
Comment (tonyr59h) [inline WordPress/src/main/res/values/strings.xml]: We should rename this to `error_media_parse_format`

File: WordPress/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -1089,6 +1090,10 @@
     <string name="error_notif_q_action_approve">Could not approve comment. Please try again later.</string>
     <string name="error_notif_q_action_reply">Could not reply to comment. Please try again later.</string>
 
+    <!-- Post Error -->
+    <string name="error_unknown_post">Unknown post, local copy of this post could be desynchronized with you server</string>
```
Comment (tonyr59h) [inline WordPress/src/main/res/values/strings.xml]: Typo, should be `...desynchronized with your server`

Comment (tonyr59h) [inline WordPress/src/main/res/values/strings.xml]: Actually, I think for English speakers `Unknown post, local copy of this post may be out of sync with the server` might make more sense.

File: WordPress/src/main/res/values/strings.xml (hunk 1 above)
Comment (maxme) [inline WordPress/src/main/res/values/strings.xml]: Do you mean `error_media_parse_error` (or just drop the `format`)?

Comment (tonyr59h) [inline WordPress/src/main/res/values/strings.xml]: Heh, yes that.

**Comment by tonyr59h** [comment]
👍  to :shipit: once Travis is 💚

**Pull request merged**